lab-kaylyn/test/portfolio-test.js

```diff
@@ -0,0 +1,165 @@
+          done();
+        });
+      });
+      describe('testing POST with invalid id', function(){
+        it('should return 404 response for invalid POST', done => {
+          request.post(`${URL}/api/portfolio/666`)
+          .end((err,res) => {
+            expect(res.status).to.equal(404);
+            done();
+          });
+        });
+      });
+    });
+  });
+  describe('testing PUT requests to  /api/portfolio', function(){
+    describe('valid id and body', function(){
+      before( done => {
+        Portfolio.createPortfolio(examplePortfolio)
+        .then( portfolio => {
+          this.tempPortfolio = portfolio;
+          done();
+        })
+        .catch(err => done(err));
+      });
+      after( done => {
+        if(this.tempPortfolio){
+          Portfolio.deletePortfolio(this.tempPortfolio.id)
+          .then(() => done())
+          .catch(done);
+        }
+      });
+      it('should return an updated portfolio via PUT', done => {
+        let updateData = {about: 'update', projects: 'updated', work: 'also updated'};
+        request.put(`${URL}/api/portfolio/${this.tempPortfolio.id}`)
+        .send(updateData)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.id).to.equal(this.tempPortfolio.id);
+          for(var key in updateData) {
+            expect(res.body[key]).to.equal(updateData[key]);
+          }
+          done();
+        });
+      });
+      it('should return a 400 for invalid PUT request', done => {
+        let updateData = 'justastring';
+        request.put(`${URL}/api/portfolio/${this.tempPortfolio.id}`)
+        .set('Content-Type', 'application/json')
+        .send(updateData)
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+  describe('testing DELETE requests to /api/portfolio', function(){
+    describe('with valid id', function(){
+      before(done => {
+        Portfolio.createPortfolio(examplePortfolio)
+        .then(portfolio => {
+          this.tempPortfolio = portfolio;
+          done();
+        })
+        .catch(err => done(err));
+      });
+      after(done => {
+        Portfolio.deletePortfolio(this.tempPortfolio.id)
+        .then(() => done())
+        .catch(err => done(err));
+      });
+      it('should return a status code of 204 upon DELETE', done => {
+        request.delete(`${URL}/api/portfolio/${this.tempPortfolio.id}`)
+        .end((err, res) => {
+          expect(res.status).to.equal(204);
+          done();
+        });
+      });
+      describe('testing DELETE requests without id', function(){
+        it('should return a status code of 404 for no id', done => {
+          request.delete(`${URL}/api/portfolio/`)
+          .end((err, res) => {
+            expect(res.status).to.equal(404);
+            done();
+          });
+        });
+      });
+    });
+  });
+});
```

Comment:
@kaylynyuh LOTS of tests!  AWESOME!  No nitpicks here, keep focusing on the TDD side of things and learning to write tests will get easier and not as much of a headache